src/app/app.component.spec.ts

```diff
@@ -1,35 +1,60 @@
-import { TestBed, async } from '@angular/core/testing';
+import {
+  ComponentFixture,
+  TestBed,
+  tick,
+  fakeAsync,
+} from '@angular/core/testing';
+
+import { Location } from '@angular/common';
+
+import { Router, RouterLinkWithHref } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
+import { routes } from './app-routing.module';
+
+import { AppModule } from './app.module';
 import { AppComponent } from './app.component';
+import { SignInComponent } from './sign-in/sign-in.component';
+import { SignUpComponent } from './sign-up/sign-up.component';
 
 describe('AppComponent', () => {
-  beforeEach(async(() => {
+  let component: AppComponent;
+  let fixture: ComponentFixture<AppComponent>;
+  let router: Router;
+  let location: Location;
+
+  beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [
-        RouterTestingModule
-      ],
-      declarations: [
-        AppComponent
-      ],
+      imports: [AppModule, RouterTestingModule.withRoutes(routes)],
```

Comment:
AppCompontent doesn't require AppModule it require it's declarations, so you can provide mocked components in declarations section in order to speed up tests